packages/plugins/apollo-managed-federation/src/index.ts

```diff
@@ -27,64 +25,91 @@ export type ManagedFederationPluginOptions = (
 };
 
 export function useManagedFederation(options: ManagedFederationPluginOptions = {}): Plugin {
-  const {
-    supergraphManager = new SupergraphSchemaManager(options as SupergraphSchemaManagerOptions),
-  } = options;
+  let _supergraphManager: SupergraphSchemaManager;
+
+  function ensureSupergraphManager() {
+    if (_supergraphManager == null) {
+      if (options instanceof SupergraphSchemaManager) {
+        _supergraphManager = options;
+      } else {
+        _supergraphManager = new SupergraphSchemaManager(options);
+      }
+    }
+    return _supergraphManager;
+  }
 
   const plugin: Plugin = {
     onYogaInit({ yoga }) {
-      supergraphManager.on('log', ({ level, message, source }) => {
-        yoga.logger[level](
-          `[ManagedFederation]${source === 'uplink' ? ' <UPLINK>' : ''} ${message}`,
-        );
-      });
+      ensureSupergraphManager().addEventListener(
+        'log',
+        ({ detail: { level, message, source } }: SupergraphSchemaManagerLogEvent) => {
```

Comment:
The goal of using an EventEmitter (implemented by ourself to avoid relying on Node specific stuff), was precisely to have a typed interface for event handlers.

Why are we moving out from it ?

Reply:
It now uses `EventTarget` which is platform agnostic, and natively available in all envs, and our `TypedEventTarget` implementation which makes it more type-safe.

Reply:
I'm not sure to understand why we changed it ? It was already platform agnostic ?

And it was more type safe, the type of the event handlers was directly I fered from the event type. Which is no longer the case here.

And we had nicer API like `once` and stuff like that.

Reply:
It was not implementing the same API that Node has for EventEmitter and it was a bit confusing to have an interface name but partially.
But in this way, we use the environment's own EventTarget implementation, and it follows the standards of handling listeners etc. So our impl there was a kind of ponyfill that needs to be tested, maintained etc seperately. Now we just use the existing EventTarget that eliminates this.